tests/unittests/test_common.h

```diff
@@ -0,0 +1,476 @@
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+/*
+ * File is intended to C and CPP compliant hence any CPP specic changes
+ * should be added into CPP section
+ *
+ */
+
+#ifdef __cplusplus
+    #include <iostream>
+    #include <iomanip>
+    #if __CUDACC__
+        #include <sys/time.h>
+    #else
+        #include <chrono>
+    #endif
+#endif
+
+// ************************ GCC section **************************
+#include <stddef.h>
+
+#include "hip/hip_runtime.h"
+#include "hip/hip_runtime_api.h"
+
+#define HC __attribute__((hc))
+
+
+#define KNRM "\x1B[0m"
+#define KRED "\x1B[31m"
+#define KGRN "\x1B[32m"
+#define KYEL "\x1B[33m"
+#define KBLU "\x1B[34m"
+#define KMAG "\x1B[35m"
+#define KCYN "\x1B[36m"
+#define KWHT "\x1B[37m"
+
+#define passed()                                                                                   \
+    printf("%sPASSED!%s\n", KGRN, KNRM);                                                           \
+    exit(0);
+
+#define failed(...)                                                                                \
+    printf("%serror: ", KRED);                                                                     \
+    printf(__VA_ARGS__);                                                                           \
+    printf("%s\n",KNRM);                                                                                  \
+    return false;
+
+#define warn(...)                                                                                  \
+    printf("%swarn: ", KYEL);                                                                      \
+    printf(__VA_ARGS__);                                                                           \
+    printf("\n");                                                                                  \
+    printf("warn: TEST WARNING\n%s", KNRM);
+
+#define skipped() printf("%sSkipped subtest %s%s\n",KYEL,__FUNCTION__,KNRM);
+
+#define HIPCHECK(error)                                                                            \
+    {                                                                                              \
+        hipError_t localError = error;                                                             \
+        if ((localError != hipSuccess) && (localError != hipErrorPeerAccessAlreadyEnabled)) {      \
+            printf("%serror: '%s'(%d) from %s at %s:%d%s\n", KRED, hipGetErrorString(localError),  \
+                   localError, #error, __FUNCTION__, __LINE__, KNRM);                                  \
```

Comment:
Can't this be merged into the ```failed``` line similar to ```HIPASSERT```?

Reply:
done